Doc/using/configure.rst

```diff
@@ -859,6 +870,13 @@ Linker flags
    :envvar:`CFLAGS_NODIST`.  Use it when a linker flag should *not* be part of
    the distutils :envvar:`LDFLAGS` once Python is installed (:issue:`35257`).
 
+   In particular, :envvar:`LDFLAGS` should not contain:
+
+   * the compiler flag `-L` (for setting the search path for libraries).
+     The `-L` flags are processed from left to right, and any flags in
+     :envvar:`LDFLAGS` would take precedence over user- and package-supplied `-L`
+     flags.
```

Comment:
In https://github.com/pyenv/pyenv/issues/2381, we found out that a `-L` flags from `LDFLAGS` was required when linking against Python to find `libintl`. We even had to add it to `python-config --ldflags` via a hack since `python-config` ignored compile-time `LDFLAGS` for some reason.

Reply:
Do you think we should add guidance here for downstream packagers that informs them of other, more suitable ways to set flags -- namely setting environment variables `CFLAGS`, `LDFLAGS`, `CPATH`, `LIBRARY_PATH` at the runtime of Python?

Reply:
> setting environment variables `CFLAGS`, `LDFLAGS`, `CPATH`, `LIBRARY_PATH` at the runtime of Python?

I'm not sure how you expect downstream packagers to do that -- and why they need to do that in the first place since `distutils` and `python-config` specifically exist to take care of providing all the necessary flags to link against a specific Python installation.

Reply:
Aren't the `pyenv` shims already setting environment variables? Specifically, they are setting `PATH`. https://trac.sagemath.org/ticket/29285#comment:20

Reply:
> I'm not sure how you expect downstream packagers to do that

conda and nix use activation scripts for their environments. homebrew provides `brew sh` for this purpose.

Reply:
That may be true for `CFLAGS`, but definitely not for `CPATH`. By the way, https://github.com/pypa/setuptools/blob/main/docs/userguide/ext_modules.rst now has some info (from my PR https://github.com/pypa/setuptools/pull/3368); we can expand it to cover all necessary details to resolve this discussion.

Reply:
This is now https://github.com/pypa/setuptools/pull/3387